Sources/FoundationEssentials/FileManager/FileOperations+Enumeration.swift

```diff
@@ -166,8 +166,13 @@ struct _FTSSequence: Sequence {
                 return
             }
 
-            state = [UnsafeMutablePointer(mutating: path), nil].withUnsafeBufferPointer { dirList in
-                guard let stream = fts_open(dirList.baseAddress!, opts, nil) else {
+            state = [Optional(UnsafeMutablePointer(mutating: path)), nil].withUnsafeBufferPointer { dirList in
+#if canImport(Android)
+                let baseAddress = unsafeBitCast(dirList.baseAddress!, to: UnsafePointer<UnsafeMutablePointer<CChar>>.self)
```

Comment:
Could you elaborate a bit more on what error this change resolves? I would have assumed that the above optional wrapping wouldn't be necessary since it'd be inferred by the compiler, and it seems that this `unsafeBitCast` of an optional pointer to a non-optional pointer would be a trap waiting for a caller that tries to read that final `nil` element, right?

Reply:
This is needed because the Android NDK has a nullability annotation on `fts_open`:

```
FTS* _Nullable fts_open(char* _Nonnull const* _Nonnull __path, int __options, int (* _Nullable __comparator)(const FTSENT* _Nonnull * _Nonnull  __lhs, const FTSENT* _Nonnull * _Nonnull __rhs)) __INTRODUCED_IN(21);
```

This makes the path pointer nonnullable . Therefore the type inference is then unable to infer the type for the array, and thus I need to explicitly make the first element optional, so that the second one can be null. Then the pointer need to be cast to match the type, as we still want to have a nil element in the array, but have non null pointer types.

Reply:
In general this stems from the fact that this specific nullability annotation in the NDK is probably incorrect, as `fts_open` is documented to take an array that has NULL as the last element.

Reply:
I added a comment in source.

Reply:
This change should no longer be needed after swiftlang/swift#74829, as [noted on your earlier commit weeks ago](https://github.com/apple/swift-corelibs-foundation/pull/4889#discussion_r1664581775).

Reply:
Good point, I need to retry with that change.

Reply:
Note [the additional linux change that I mentioned there](https://github.com/swiftlang/swift/pull/74829#discussion_r1681611802), we'll need to do something like that for that change to register on the linux CI.

Reply:
Oh wait, that pull isn't needed at all on the linux CI, as this function signature only varies in Bionic.

In that case, this pull can just be closed, if #714 gets macros working and another small apinotes pull avoids the need for this change.